packages/devtools_app/lib/src/flutter/status_line.dart

```diff
@@ -2,26 +2,225 @@
 
 import '../../devtools.dart' as devtools;
+import '../globals.dart';
+import '../service_manager.dart';
+import 'common_widgets.dart';
+import 'notifications.dart';
+import 'screen.dart';
+
+const statusLineHeight = 24.0;
 
 /// The status line widget displayed at the bottom of DevTools.
+///
+/// This displays information global to the application, as well as gives pages
+/// a mechanism to display page-specific information.
 class StatusLine extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    // TODO(devoncarew): Break this into an isolates area, a connection status
-    // area, a page specific area, and a documentation link area.
-
     final textTheme = Theme.of(context).textTheme;
 
+    final Screen currentScreen = Provider.of<Screen>(context);
+
+    final List<Widget> children = [];
+
+    // Have an area for page specific help (always docked to the left).
+    children.addAll([
+      Expanded(
+        child: Align(
+          alignment: Alignment.centerLeft,
+          child: buildHelpUrlStatus(context, currentScreen, textTheme),
+        ),
+      ),
+      BulletSpacer(),
+    ]);
+
+    // Optionally display page specific status.
+    if (currentScreen != null && currentScreen.providesStatus) {
+      children.addAll([
+        Expanded(
+          child: Align(
+            alignment: Alignment.center,
+            child: buildPageStatus(context, currentScreen, textTheme),
+          ),
+        ),
+        BulletSpacer(),
+      ]);
+    }
+
+    // Optionally display an isolate selector.
+    if (currentScreen != null && currentScreen.usesIsolateSelector) {
+      children.addAll([
+        Expanded(
+          child: Align(
+            alignment: Alignment.center,
+            child: buildIsolateSelector(context, textTheme),
+          ),
+        ),
+        BulletSpacer(),
+      ]);
+    }
+
+    // Always display connection status (docked to the right).
+    children.addAll([
+      Expanded(
+        child: Align(
+          alignment: Alignment.centerRight,
+          child: buildConnectionStatus(textTheme),
+        ),
+      ),
+    ]);
+
     return Container(
-      height: 24.0,
+      height: statusLineHeight,
       alignment: Alignment.centerLeft,
-      child: Text(
-        'DevTools ${devtools.version}',
-        style: textTheme.bodyText2,
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: children,
       ),
     );
   }
+
+  Widget buildHelpUrlStatus(
+    BuildContext context,
+    Screen currentScreen,
+    TextTheme textTheme,
+  ) {
+    final String docPageId = currentScreen.docPageId;
+    if (docPageId != null) {
+      return InkWell(
+        onTap: () async {
+          // TODO(devoncarew): Shorten these urls to something like
+          // 'https://flutter.dev/devtools/$docPageId'.
+          final url =
+              'https://flutter.dev/docs/development/tools/devtools/$docPageId';
```

Comment:
@sfshaza2 - do you think we could make the direct URLs to the DevTools' docs shorter? We're going to be showing these to users. Instead of something like:

```
https://flutter.dev/docs/development/tools/devtools/inspector
```

it would be great to have a url like:

```
https://flutter.dev/devtools/inspector
```

The docs could either be hosted at that url, or I think it would be OK if they re-directed to the longer urls (https://flutter.dev/devtools/* ==> https://flutter.dev/docs/development/tools/devtools/*).

Reply:
@devoncarew, I don't want to physically move the files, but we can *definitely* set up redirects like that in the firebase.json file at the top of the repo!

Reply:
Also, by "showing the users" do you mean from DevTools? Any reason the tooling can't provide an actual link?